whois-api/src/main/java/net/ripe/db/whois/api/rdap/WhoisRdapService.java

```diff
@@ -117,41 +110,31 @@ public WhoisRdapService(final RdapQueryHandler rdapQueryHandler,
     @Produces({MediaType.APPLICATION_JSON, CONTENT_TYPE_RDAP_JSON})
     @Path("/{objectType}/{key:.*}")
     public Response lookup(@Context final HttpServletRequest request,
-                           @PathParam("objectType") final String objectType,
+                           @PathParam("objectType") RdapRequestType requestType,
                            @PathParam("key") final String key) {
 
         LOGGER.info("Request: {}", RestServiceHelper.getRequestURI(request));
+        Set<ObjectType> whoisObjectTypes = requestType.getWhoisObjectTypes(key);
 
-        switch (objectType.toLowerCase()) {
-            case "autnum": {
+        switch (requestType) {
+            case autnum: {
                 String autnumKey = String.format("AS%s", key);
                 rdapRequestValidator.validateAutnum(autnumKey);
                 return lookupForAutNum(request, autnumKey);
             }
-            case "domain": {
+            case domain: {
                 rdapRequestValidator.validateDomain(key);
-                final Set<ObjectType> whoisObjectTypes = Sets.newHashSet();
-                whoisObjectTypes.add(DOMAIN);
                 return lookupObject(request, whoisObjectTypes, key);
             }
-            case "ip": {
+            case ip: {
                 rdapRequestValidator.validateIp(request.getRequestURI(), key);
-                return lookupWithRedirectUrl(request, key.contains(":") ? INET6NUM : INETNUM, key);
+                return lookupWithRedirectUrl(request, whoisObjectTypes, key);
             }
-            case "entity": {
+            case entity: {
                 rdapRequestValidator.validateEntity(key);
-
-                final Set<ObjectType> whoisObjectTypes = Sets.newHashSet();
-                if (key.toUpperCase().startsWith("ORG-")) {
-                    whoisObjectTypes.add(ORGANISATION);
-                } else {
-                    whoisObjectTypes.add(PERSON);
-                    whoisObjectTypes.add(ROLE);
-                    whoisObjectTypes.add(MNTNER);
-                }
                 return lookupObject(request, whoisObjectTypes, key);
             }
-            case "nameserver": {
+            case nameserver: {
                 throw rdapExceptionMapper.notFound("nameserver not found");
```

Comment:
Why have all exceptions handled in the normal manner by the RdapExceptionMapper but for this case have the exceptionmapper build an exception which wraps the 404 and then throw this exception which is then caught by the exeptionhandler again as a WebApplicationException and just passed through?

Reply:
This was resolved.